Automattic/jetpack
# Search: hide unsupported taxonomies from widget

Fixes #24692 

#### Changes proposed in this Pull Request:
<strike>

- Adds custom taxonomies: 
  - 'oecm_audience'
  - 'tspa-resources-topic'
  - 'tspa-resources-type'
  - 'tspa-curriculum-type'
- Added core taxonomies
  - 'category'
  - 'post_tag' 
  - 
</strike>

The PR Hides taxonomies that doesn't exists in our syned taxonomy list, and introduces a filter `jetpack_search_allowed_taxonomies_for_widget_filters` to allow user manage taxonomies shown in search widget filter drop down.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Add the following snippet in active theme's `functions.php`
```PHP
register_taxonomy(
	'foobar-resources-topic',
	'post',
	array(
		'label'             => __( 'Foobar Topic', 'foobar' ),
		'description'       => __( 'The resource topic to be grouped under.', 'foobar' ),
		'public'            => true,
		'hierarchical'      => true,
		'show_admin_column' => true,
	)
);

```
- Switch to branch `fix/hide-unsupported-taxonomies-from-widget-filters`
- Ensure 'Foobar Topic' is NOT shown is JP search widget Taxonomy dropdown


![image](https://user-images.githubusercontent.com/1425433/175170240-6c041174-dd85-4bae-a5c6-b2792461eb70.png)


- Add the following snippet in active theme's `functions.php`
```PHP
add_filter( 'jetpack_search_allowed_taxonomies_for_widget_filters', function( $taxonomies ){
    $taxonomies[] = 'foobar-resources-topic';
    return $taxonomies;
} );
```
- Ensure 'Foobar Topic' is shown is JP search widget Taxonomy dropdown

![image](https://user-images.githubusercontent.com/1425433/175169539-c0e02b63-89bd-4e4a-93de-18ea4bc9a454.png)
